internal/datasourcev2/streaming_data_source.go

```diff
@@ -249,36 +242,29 @@ func (sp *StreamProcessor) consumeStream(stream *es.Stream, closeWhenReady chan<
 				currentChangeSet = changeSet{events: make([]es.Event, 0)}
 				//nolint: godox
 				// TODO: Do we need to restart here?
-			case "payload-transferred":
+			case fdv2proto.EventPayloadTransferred:
 				currentChangeSet.events = append(currentChangeSet.events, event)
-				updates, err := processChangeset(currentChangeSet)
+				updates, err := deserializeEvents(currentChangeSet.events)
 				if err != nil {
 					sp.loggers.Errorf("Error processing changeset: %s", err)
 					gotMalformedEvent(nil, err)
 					break
 				}
-				for _, update := range updates {
-					switch u := update.(type) {
-					case datasource.PatchData:
-						if !sp.dataDestination.Upsert(u.Kind, u.Key, u.Data) {
-							storeUpdateFailed("streaming update of " + u.Key)
-						}
-					case datasource.PutData:
-						if sp.dataDestination.Init(u.Data, nil) {
-							sp.setInitializedAndNotifyClient(true, closeWhenReady)
-						} else {
-							storeUpdateFailed("initial streaming data")
-						}
-					case datasource.DeleteData:
-						deletedItem := ldstoretypes.ItemDescriptor{Version: u.Version, Item: nil}
-						if !sp.dataDestination.Upsert(u.Kind, u.Key, deletedItem) {
-							storeUpdateFailed("streaming deletion of " + u.Key)
-						}
-
-					default:
-						sp.loggers.Infof("Unexpected update found in changeset: %s", update)
+
+				//nolint:godox
+				// TODO(SDK-712): If the destination fails to apply the updates should it affect the processor?
+				// This would only happen in the case of a persistent store.
+
+				switch currentChangeSet.intent.Payloads[0].Code {
+				case fdv2proto.IntentTransferFull:
+					{
+						_ = sp.dataDestination.SetBasis(updates, fdv2proto.NoSelector(), true)
```

Comment:
Shouldn't return errors here. Should handle in data system, then use DataStoreStatus or other mechanism for testing / reporting